rectools/models/rank.py

```diff
@@ -149,18 +151,23 @@ def _rank_on_gpu(
         object_norms: tp.Optional[np.ndarray],
         filter_query_items: tp.Optional[tp.Union[sparse.csr_matrix, sparse.csr_array]],
     ) -> tp.Tuple[np.ndarray, np.ndarray]:  # pragma: no cover
-        object_factors = implicit.gpu.Matrix(object_factors.astype(np.float32))
+
+        def _convert_arr_to_implicit_gpu_matrix(arr: np.ndarray) -> implicit.gpu.Matrix:
```

Comment:
We also create `gpu.Matrix` in the ALSWrapper code

Let's please make this function an util and use everywhere

Reply:
fixed

Reply:
I remembered now why I didn't do this as a public function. When implicit is built without cuda, it cannot import it because of the return type annotation. Wo we really need it?

Reply:
No, I'm not saying we need this public. I just want to add this logic with copy wherever we create gpu.Matrix

Reply:
fixed